server/build_event_protocol/build_status_reporter/build_status_reporter.go

```diff
@@ -49,6 +49,11 @@ func NewBuildStatusReporter(env environment.Env, buildEventAccumulator *accumula
 }
 
 func (r *BuildStatusReporter) ReportStatusForEvent(ctx context.Context, event *build_event_stream.BuildEvent) {
+	if r.githubClient.WorkflowID() != r.buildEventAccumulator.WorkflowID() {
+		r.githubClient.SetWorkflowID(r.buildEventAccumulator.WorkflowID())
+	}
```

Comment:
Trying to follow this -- so during the BEStream you'll get a bunch of events, on each one you'll call this, eventually at some point this code will set the workflow ID on githubClient -- how do you know that you haven't already published a status or something previously?

Reply:
Happy to approve this as-is, but in my experience, these kinds of things can come back to bite you :P I'm not sure the best way to correctly handle it in this case -- maybe by not creating the github client until we're sure we've gotten the auth token we need? Anyway, this is fine for an mvp :)